### DISCUSSION TITLE
TotalCount for Cursor Pagination

### DISCUSSION THREAD

**Discussion by leonardocustodio** [laravel/framework#43045] [Ideas]
Hello,

One very common scenario for using cursor pagination for APIs nowadays is to implement GraphQL connections: https://relay.dev/graphql/connections.htm - there is a spec for this and unfortunately, the only thing that we cannot do to implement this spec when using the "cursorPaginate" is returning the totalCount.
I would like to ask if it would be possible to add this parameter or if the only way to go would be to make two different queries.

Best regards,
Leonardo

**Comment by leonardocustodio** [comment]
I've sent a PR for this if you guys find suitable: https://github.com/laravel/framework/pull/43046

**Comment by cwilby** [comment]
Added some thoughts here
https://github.com/laravel/framework/pull/43046#issuecomment-1174407886

TL;DR - cursor pagination is used for other reasons than scaling.  It solves issues relating to synchronizing client-side paginated data.  Having a count _could_ be useful in that scenario.

It would have to be turned off by default, as most will turn to cursor pagination for large datasets.

**Reply by leonardocustodio** [reply]
We also use for the same reason as yours, not for large datasets but because there are items being added in real-time so we use it to not repeat the items when changing the page. Anyway I closed so we can discuss a bit further and on how we could implement this if everyone agrees instead of going for the PR directly since the PR added it as the default behavior and as suggested it should not be the default. Thanks for your notes, nice to know I'm not the only one 😁

**Reply by cwilby** [reply]
I think a reasonable path is to create another PR with default functionality baked in, and a cohesive explanation as to why this is needed, and where it could be used.

I am not building my example with the Laravel framework, but if I were this would help to standardize the solution.
